### PR TITLE
Add create account CRUD

### PR DESCRIPTION
**Pull request by jadmnsaad**
https://apps.nrs.gov.bc.ca/int/jira/browse/EPICSYSTEM-208

- Add create account resource
- Add request schemas
- Add services
- Add model methods

### CODE REVIEW

**Review by davidjnunezf — APPROVED** [review]

File: submit-api/migrations/versions/97de805275ec_.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.bulk_insert(
+        roles_table,
+        [
+            {
+                'role_name': 'ACCOUNT_PRIMARY_ADMIN',
+                'description': 'Administrator role',
+                'created_date': datetime.utcnow(),
+                'created_by': 'system'  # Or replace with the actual creator
```
Comment (dinesh-aot) [inline submit-api/migrations/versions/97de805275ec_.py:41]: What do you think of adding rest of the roles if you already know them like PROJECT_ADMIN

File: submit-api/src/submit_api/models/account.py [hunk 2]
```diff
@@ -16,4 +16,22 @@ class Account(BaseModel):
...
+        """Fetch account by proponent id."""
+        return cls.query.filter_by(proponent_id=proponent_id).first()
+
+    @classmethod
+    def create_account(cls, account_data, session=None) -> Account:
+        """Create account."""
+        account = Account(
+            proponent_id=account_data.get('proponent_id', None),
+        )
+        if session:
```
Comment (dinesh-aot) [inline submit-api/src/submit_api/models/account.py:32]: May I know the purpose of this checking "if session"

Comment (jadmnsaad) [reply submit-api/src/submit_api/models/account.py:32]: if we have 3 transactions and one fail and we need to rollback everything if one fails I implemented the above pattern.

you pass the same session for all 3. if no session is passed then it's just db.session.commit or something

File: submit-api/migrations/versions/97de805275ec_.py (hunk 1 above)
Comment (jadmnsaad) [reply submit-api/migrations/versions/97de805275ec_.py:41]: let's agree on the list and I can follow that up in another PR

File: submit-api/src/submit_api/models/account.py (hunk 2 above)
Comment (jadmnsaad) [reply submit-api/src/submit_api/models/account.py:32]: image

Comment (dinesh-aot) [reply submit-api/src/submit_api/models/account.py:32]: I got it. I hope we will be using it only  in those scenarios where there are multiple database transactions involved.

**Review by dinesh-aot — APPROVED** [review]

**Pull request merged**